src/test/scala/mesosphere/marathon/tasks/TaskIdTest.scala

```diff
@@ -27,20 +28,20 @@ class TaskIdTest extends UnitTest {
     }
 
     "Old TaskIds can be converted even if they have underscores in them" in {
-      val taskId = Task.Id(TaskID.newBuilder().setValue("app_foo_bar_0-12345678").build)
+      val taskId = Task.Id(TaskID.newBuilder().setValue("app_foo_bar_682ebe64-0771-11e4-b05d-e0f84720c54e").build)
       taskId.runSpecId should equal("/app/foo/bar".toRootPath)
     }
 
     "TaskIds with encoded InstanceIds could be encoded" in {
-      val taskId = Task.Id(TaskID.newBuilder().setValue("test_foo_bla_rest.instance-instance1.62d0f03f-79aa-11e6-a1a0-660c139c5e15").build)
+      val taskId = Task.Id(TaskID.newBuilder().setValue("test_foo_bla_rest.instance-62d0f03f-79aa-11e6-a1a0-660c139c5e15.$anon").build)
```

Comment:
Here as well. The uuid would be `instance1.62d0f03f-79aa-11e6-a1a0-660c139c5e15` and it would be a `LegacyId` as far as I can tell.